docs/guides/prefect-deploy.md

```diff
@@ -0,0 +1,628 @@
+
+For example, a user wishing to store their code in an S3 bucket and rely on default worker settings for its runtime environment could use the `s3` recipe:
+
+<div class="terminal">
+```bash
+$ prefect init --recipe s3
+>> bucket: < insert bucket name here >
+```
+</div>
+
+Inspecting our newly created `prefect.yaml` file we find that the `push` and `pull` sections have been templated out for us as follows:
+
+```yaml
+push:
+  - prefect_aws.deployments.steps.push_to_s3:
+      id: push-code
+      requires: prefect-aws>=0.3.0
+      bucket: my-bucket
+      folder: project-name
+      credentials: null
+
+pull:
+  - prefect_aws.deployments.steps.pull_from_s3:
+      requires: prefect-aws>=0.3.0
+      bucket: my-bucket
+      folder: "{{ push-code.folder }}"
+      credentials: null
+```
+
+The bucket has been populated with our provided value (which also could have been provided with the `--field` flag); note that the `folder` property of the `push` step is a template - the `pull_from_s3` step outputs both a `bucket` value as well as a `folder` value that can be used to template downstream steps.  Doing this helps you keep your steps consistent across edits.
+
+As discussed above, if you are using [blocks](/concepts/blocks/), the credentials section can be templated with a block reference for secure and dynamic credentials access:
+
+```yaml
+push:
+  - prefect_aws.deployments.steps.push_to_s3:
+      requires: prefect-aws>=0.3.0
+      bucket: my-bucket
+      folder: project-name
+      credentials: "{{ prefect.blocks.aws-credentials.dev-credentials }}"
+```
+
+Anytime you run `prefect deploy`, this `push` section will be executed upon successful completion of your `build` section. For more information on the mechanics of steps, [see below](#deployment-mechanics).
+
+### The Pull Action
+
+The pull section is the most important section within the `prefect.yaml` file as it contains instructions for preparing your flows for a deployment run.  These instructions will be executed each time a deployment created within this folder is run via a worker.
+
+There are three main types of steps that typically show up in a `pull` section:
+
+- `set_working_directory`: this step simply sets the working directory for the process prior to importing your flow
+- `git_clone`: this step clones the provided repository on the provided branch
+- `pull_from_{cloud}`: this step pulls the working directory from a Cloud storage location (e.g., S3)
+
+!!! tip "Use block and variable references"
+    All [block and variable references](#templating-options) within your pull step will remain unresolved until runtime and will be pulled each time your deployment is run. This allows you to avoid storing sensitive information insecurely; it also allows you to manage certain types of configuration from the API and UI without having to rebuild your deployment every time.
```

Comment:
```suggestion
    All [block and variable references](#templating-options) within your pull step will remain unresolved until runtime and will be pulled each time your deployment is run. 
    This allows you to avoid storing sensitive information insecurely.
    It also allows you to manage certain types of configuration from the API and UI without having to rebuild your deployment every time.
```